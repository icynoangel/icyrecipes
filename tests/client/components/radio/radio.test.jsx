import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import Radio from './../../../../src/client/js/components/radio/radio';
import renderer from 'react-test-renderer';

describe('<Radio />', function() {
  this.getComponent = (checked = false, disabled = false) => {
    return (
      <Radio
        onClick={this.onClick}
        label={this.label}
        value={this.value}
        checked={checked}
        disabled={disabled}
      />
    );
  };

  beforeEach(() => {
    this.onClick = jest.fn();
    this.label = 'Radio Button';
    this.value = 'rad_val';
  });

  it('Should render <Radio />', () => {
    const radio = renderer.create(this.getComponent()).toJSON();
    expect(radio).toMatchSnapshot();
  });

  it('Should render a checked <Radio />', () => {
    const radio = renderer.create(this.getComponent(!!'checked')).toJSON();
    expect(radio).toMatchSnapshot();
  });

  it('Should render a disabled <Radio />', () => {
    const radio = renderer
      .create(this.getComponent(!'checked', !!'disabled'))
      .toJSON();
    expect(radio).toMatchSnapshot();
  });

  it('Should render a checked and disabled <Radio />', () => {
    const radio = renderer
      .create(this.getComponent(!!'checked', !!'disabled'))
      .toJSON();
    expect(radio).toMatchSnapshot();
  });

  it('Should call onClick prop', () => {
    const wrapper = shallow(this.getComponent());
    wrapper.find('.radio').simulate('click');
    expect(this.onClick).toHaveBeenCalledTimes(1);
    expect(this.onClick).toHaveBeenCalledWith(this.value);
  });

  it('Should not call onClick prop when it is disabled', () => {
    const wrapper = shallow(this.getComponent(!'checked', !!'disabled'));
    wrapper.find('.radio').simulate('click');
    expect(this.onClick).not.toHaveBeenCalled();
  });
});
