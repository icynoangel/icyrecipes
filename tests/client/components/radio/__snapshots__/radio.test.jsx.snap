// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> Should render <Radio /> 1`] = `
<div
  className="radio"
  onClick={[Function]}
>
  <div
    className="radio__button"
  >
    <div
      className="radio__button__checked"
    />
  </div>
  <div
    className="radio__label"
  >
    Radio Button
  </div>
</div>
`;

exports[`<Radio /> Should render a checked <Radio /> 1`] = `
<div
  className="radio --checked"
  onClick={[Function]}
>
  <div
    className="radio__button"
  >
    <div
      className="radio__button__checked"
    />
  </div>
  <div
    className="radio__label"
  >
    Radio Button
  </div>
</div>
`;

exports[`<Radio /> Should render a checked and disabled <Radio /> 1`] = `
<div
  className="radio --checked --disabled"
  onClick={[Function]}
>
  <div
    className="radio__button"
  >
    <div
      className="radio__button__checked"
    />
  </div>
  <div
    className="radio__label"
  >
    Radio Button
  </div>
</div>
`;

exports[`<Radio /> Should render a disabled <Radio /> 1`] = `
<div
  className="radio --disabled"
  onClick={[Function]}
>
  <div
    className="radio__button"
  >
    <div
      className="radio__button__checked"
    />
  </div>
  <div
    className="radio__label"
  >
    Radio Button
  </div>
</div>
`;
